motr/idx_dix.c

```diff
@@ -549,16 +549,25 @@ static void dix_build(const struct m0_op_idx *oi,
 {
 	unsigned int   opcode = OP_IDX2CODE(oi);
 	struct m0_idx *idx = oi->oi_idx;
+	int            rc;
 
 	M0_SET0(out);
 	out->dd_fid = *OI_IFID(oi);
 	/* Pool version and layout type which are passed by consumers like S3 */
 	if (M0_IN(opcode, (M0_IC_GET, M0_IC_PUT, M0_IC_DEL, M0_IC_NEXT))) {
-		if (idx->in_attr.idx_layout_type == DIX_LTYPE_DESCR) {
-			M0_LOG(M0_DEBUG, "Opcode: %u, DIX pool version:"FID_F"",
+		if ((idx->in_attr.idx_layout_type == DIX_LTYPE_DESCR) 
```

Comment:
Shall we have the "struct m0_idx_attr" updated as:
```
 766 struct m0_idx_attr {
 767         /** DIX pool layout type. */
 768         enum dix_layout_type  idx_layout_type;
 769         /** DIX pool version. */
 770         struct m0_fid idx_pver;
```

Reply:
As per design, we have to export pool version and layout type to consumers like S3. I believe, no need to add in "struct m0_idx_attr"

Reply:
@huanghua78 : As `idx_layout_type` is stored in s3 meta-data, i think it's better to keep it as uint32